### PR TITLE
Remove redundant constructors from some classes in `io.trino.sql.tree` package

### PR DESCRIPTION
**Pull request by ebyhr**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java [hunk 1]
```diff
@@ -255,7 +256,7 @@ public void testUnsupportedAddAmbiguousField()
 
         assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(tableName), QualifiedName.of("col", "a", "z"), INTEGER, false, false)))
                 .hasErrorCode(AMBIGUOUS_NAME)
-                .hasMessage("Field path [a, z] within row(a row(x integer), A row(y integer)) is ambiguous");
+                .hasMessageContaining("Field path [a, z] within row(a row(x integer), A row(y integer)) is ambiguous");
```
Comment (wendigo) [inline core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java:259]: this doesn't belong to this commit

File: core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java [hunk 2]
```diff
@@ -63,7 +64,7 @@ public void testCommentTableOnView()
 
         assertTrinoExceptionThrownBy(() -> getFutureValue(setComment(TABLE, asQualifiedName(viewName), Optional.of("new comment"))))
                 .hasErrorCode(TABLE_NOT_FOUND)
-                .hasMessage("Table '%1$s' does not exist, but a view with that name exists. Did you mean COMMENT ON VIEW %1$s IS ...?", viewName);
+                .hasMessageContaining("Table '%1$s' does not exist, but a view with that name exists. Did you mean COMMENT ON VIEW %1$s IS ...?", viewName);
```
Comment (wendigo) [inline core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java:67]: oh, I see it now, this now contains a location?

**Review by wendigo — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java (hunk 2 above)
Comment (ebyhr) [reply core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java:67]: Right, the message contains a dummy location at the beginning.
